### PR TITLE
Fix various UI bugs

### LINKED ISSUE #3215: Loading a PO from the project manager makes KeY freeze
## Description

Loading a PO from the project manager makes KeY freeze.

## Reproducible
  
always on my machine

### Steps to reproduce 

1. Open the example selector window
2. Select "Remove Duplicates"
3. Select the first method PO
4. Run auto to close
5. Open the proof manager
6. Select the second method PO
7. KeY freezes

### Additional information

git bisect reveals that c07a00fba07e8513966c2e958d797a5d6023d424 is the culprit.

I wonder if this removal has to do with the problem:
```diff
     @Override
     public void progressStopped(Object sender) {
-        mainWindow.getMediator().startInterface(true);
+        // no need to call startInterface(), will be done by ProblemLoader once loading is done
     }
```

---

* Commit: 0ac6d7f2f89621db5bcd223c76470fdcc857ac4d

### LINKED ISSUE #3228: FileNotFoundException because of missing proof settings
## Description

KeY throws a FileNotFoundException if the proof-settings file is missing

## Reproducible
  
always

### Steps to reproduce 

1. Delete or rename your `~/.key` directory
2. Start KeY
3. Observe the exception "No proof-settings could be loaded, using defaultsjava.io.FileNotFoundException: ~/.key/proof-settings.props (No such file or directory)" This exception is thrown whenever you start KeY until you create a proof-settings file by loading a proof obligation.

---

* Commit: 0ccf763d02de8e5b63d293d42ee214d079637bde

### PR DESCRIPTION
**Pull request by FliegendeWurst**
I recommend reviewing this PR commit-by-commit. Most of the bugs were found using the new UI testing facility (https://github.com/KeYProject/key/pull/3216/commits/3d73e59bd6ba8a8a15fb36f93aa193d3d7bd65d3).

Fixes #3228, #3227, #3215, #1648, https://github.com/KeYProject/key/issues/1652#issuecomment-1613075565, #1526, #1117 and task 1+4 of #3214

### CODE REVIEW

**Comment by mattulbrich** [comment]
#3214 could be of interest here too.

File: key.ui/src/main/java/org/key_project/util/java/SwingUtil.java [hunk 1]
```diff
@@ -149,4 +150,46 @@ public static JScrollPane createScrollPane(JTable table) {
...
+     *
+     * @param tree the tree to search
+     * @param tp the tree path
+     * @return a tree path or null
+     */
+    public static TreePath findPathInNewTree(JTree tree, TreePath tp) {
+        TreePath root = tree.getPathForRow(0);
+        Object last = root.getLastPathComponent();
+        int pathIndex = 0;
+        if (!last.toString().equals(tp.getPathComponent(pathIndex).toString())) {
```
Comment (jwiesler) [inline key.ui/src/main/java/org/key_project/util/java/SwingUtil.java:166]: Why does this have to be the toString repr? Filtering shouldn't change the nodes so why is reference equality not enough?

**Review by jwiesler — APPROVED** [review]

File: key.ui/src/main/java/org/key_project/util/java/SwingUtil.java (hunk 1 above)
Comment (FliegendeWurst) [reply key.ui/src/main/java/org/key_project/util/java/SwingUtil.java:166]: The nodes are cleared here: https://github.com/KeYProject/key/blob/0ac6d7f2f89621db5bcd223c76470fdcc857ac4d/key.ui/src/main/java/de/uka/ilkd/key/gui/prooftree/GUIProofTreeModel.java#L377-L380

Comment (FliegendeWurst) [reply key.ui/src/main/java/org/key_project/util/java/SwingUtil.java:166]: I think your approach would be more efficient. I'll use it instead if it works as expected.

**Pull request merged**